File: frontend/src/types/race.ts
export interface Race {
  id: string; // Use string or number based on backend ID type (UUID is string)
  name: string;
  // Add fields directly, making them optional as they might be null in DB
  city?: string;
  state?: string;
  lat?: number;
  lng?: number;
  date: string; // API returns date as string now
  distance?: '5K' | '10K' | 'Half Marathon' | 'Marathon' | '50K' | '50 Miles' | '100K' | '100 Miles' | 'Other'; // Updated with ultra distances
  total_elevation_gain?: number; // Added new numeric field
  website?: string;
  ai_summary?: string;
  pr_potential_score?: number;
  similar_runners_count?: number;
  training_groups_count?: number;
  similar_pace_runners_count?: number;
  image_url?: string | null;
  // Add other fields from Pydantic model if needed for type consistency
  flatness_score?: number;
  view_count?: number;
  save_count?: number;
  plan_count?: number;
  created_at?: string; // Dates from backend often come as ISO strings
  updated_at?: string;
} 

File: backend/app/models/race.py
from pydantic import BaseModel, HttpUrl, field_validator
from typing import Optional, Literal
from datetime import date, datetime
import uuid

class RaceBase(BaseModel):
    name: str
    city: Optional[str] = None
    state: Optional[str] = None
    lat: Optional[float] = None
    lng: Optional[float] = None
    distance: Optional[Literal['5K', '10K', 'Half Marathon', 'Marathon', '50K', '50 Miles', '100K', '100 Miles', 'Other']] = None
    date: Optional[str] = None
    total_elevation_gain: Optional[int] = None
    flatness_score: Optional[int] = None  # e.g., 1-5 (1=very hilly, 5=very flat)
    pr_potential_score: Optional[float] = None
    ai_summary: Optional[str] = None
    website: Optional[str] = None
    similar_runners_count: Optional[int] = None
    training_groups_count: Optional[int] = None
    similar_pace_runners_count: Optional[int] = None
    view_count: int = 0
    save_count: int = 0
    plan_count: int = 0

class RaceCreate(RaceBase):
    # Fields required on creation, if different from base
    pass

class RaceUpdate(BaseModel):
    # Optional fields for updates
    name: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    distance: Optional[Literal['5K', '10K', 'Half Marathon', 'Marathon', '50K', '50 Miles', '100K', '100 Miles', 'Other']] = None
    date: Optional[str] = None
    flatness_score: Optional[int] = None
    pr_potential_score: Optional[float] = None
    ai_summary: Optional[str] = None
    website: Optional[str] = None
    view_count: Optional[int] = None
    save_count: Optional[int] = None
    plan_count: Optional[int] = None

class Race(RaceBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True # Use Pydantic v2 standard 